pkg/server/commands/list_objects.go

```diff
@@ -270,11 +269,12 @@ func (q *ListObjectsQuery) evaluate(
 		reverseExpandResultsChan := make(chan *reverseexpand.ReverseExpandResult, 1)
 		objectsFound := atomic.Uint32{}
 
+		metricsDs := storagewrappers.NewMetricsOpenFGAStorage(q.datastore)
 		ds := storagewrappers.NewCombinedTupleReader(
-			q.datastore,
+			storagewrappers.NewBoundedConcurrencyTupleReader(
+				metricsDs, q.maxConcurrentReads),
 			req.GetContextualTuples().GetTupleKeys(),
 		)
```

Comment:
FYI i dislike how this looks. Discussed offline with @jpadilla, the current problem we have is that some datastore wrappers are server-wide, while others are request-specific.

What we could do in a future PR is to have every query object create it's own datastore wrapper _from scratch_. The server should _only_ hold any cache objects that those wrappers need.